src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -125,15 +141,31 @@ void AdaptDispatch::_SetRgbColorsHelperFromSubParams(const VTParameter colorItem
         if (tableIndex <= 255)
         {
             const auto adjustedIndex = gsl::narrow_cast<BYTE>(tableIndex);
-            if (isForeground)
-            {
-                attr.SetIndexedForeground256(adjustedIndex);
-            }
-            else
-            {
-                attr.SetIndexedBackground256(adjustedIndex);
-            }
+            color = TextColor{ adjustedIndex, true };
         }
+        break;
+    }
+    default:
+        break;
+    };
+
+    // apply the color iff it's not default.
+    if (!color.IsDefault())

```

Comment:
"if it's not the default" is obvious to me. I believe this comment would be better if it would state why avoiding `IsDefault()` is necessary.

Personally, I would write this code a little bit differently and avoid the need for the extra check using early returns in the first switch/case:
```cpp
TextColor color;

switch (typeOpt)
{
case DispatchTypes::GraphicsOptions::RGBColorOrFaint:
    if (hasColorSpaceId || red > 255 || green > 255 || blue > 255)
    {
        return;
    }
    color = TextColor{ RGB(red, green, blue) };
    break;
}
case DispatchTypes::GraphicsOptions::BlinkOrXterm256Index:
    if (tableIndex > 255)
    {
        return;
    }
    color = TextColor{ adjustedIndex, true };
    break;
default:
    return;
}

switch (colorItem)
{
case ForegroundExtended:
    attr.SetForeground(color);
    break;
case BackgroundExtended:
    attr.SetBackground(color);
    break;
case UnderlineColor:
    attr.SetUnderlineColor(color);
    break;
default:
    break;
};
```

Reply:
Just updated the code to use lambda function instead. I think it's better now, what do you think?

Reply:
Looks good to me. 🙂